doc/reference/trd-storage-permissions.md

```diff
@@ -0,0 +1,354 @@
+stored state identifier. That is, a Read permission gives an application
+permission to read only stored state marked with the associated stored state
+identifier, and a Modify permission gives an application permission to modify
+only stored state marked with the associated stored state identifier.
+
+
+5 Requirements
+-------------------------------
+
+The Tock storage model imposes the following requirements:
+
+1. Applications are given separate write, read, and modify permissions.
+2. The label stored with the persistent data when the data are written is the
+   application's short AppID.
+3. Applications without a `ShortId::Fixed` cannot access (i.e.,
+   read/write/modify) any persistent storage.
+4. How permissions are mapped to applications must be customizable for different
+   Tock kernels.
+
+Additionally, the kernel itself can be given permission to store state.
+
+### 5.1 ShortId Implications
+
+As all persistent state written by applications is marked with the writing
+application's ShortId, the assignment mechanism for ShortIds is tightly coupled
+with the access policies for persistent state. This coupling is intentional as
+AppIDs are unique to specific applications. However, as ShortIds are only 32
+bits, it is not possible to assign a globally unique ShortId to all
+applications. Therefore, board authors should be intentional with how ShortIds
+are assigned when persistent storage is accessible to userspace.
+
+In particular, two potentially problematic cases can arise:
+
+1. A ShortId is re-used for different applications. This might happen if one
+   application is discontinued and a new application is assigned the same
+   ShortId. The new application would then have unconditional access to any
+   state the old application stored.
+2. A new ShortId is used for the same application. This might happen if the
+   ShortId assignment algorithm changes. The same application then would lose
+   access to data it previously stored.
+
+
+6 Kernel Enforcement
+-------------------------------
+
+It is not feasible to implement all persistent storage APIs through the core
+kernel (i.e., in trusted code). Instead, the kernel provides an API to retrieve
+the storage permissions for a specific process. Capsules then use these
+permissions to enforce restrictions on storage access.
+The API consists of these functions:
+
+```rust
+/// Check if these storage permissions grant read access to the stored state
+/// marked with identifier `stored_id`.
+pub fn check_read_permission(&self, stored_id: u32) -> bool;
+
+/// Check if these storage permissions grant modify access to the stored
+/// state marked with identifier `stored_id`.
+pub fn check_modify_permission(&self, stored_id: u32) -> bool;
+
+/// Retrieve the identifier to use when storing state, if the application
+/// has permission to write. Returns `None` if the application cannot write.
+pub fn get_write_id(&self) -> Option<u32>;
```

Comment:
I'm not sure if I correctly understand the meaning if `store_id`. Is this the actial `short_id` that stored the data?